neuronpp/core/cells/utils.py

```diff
@@ -18,4 +18,65 @@ def get_vecstim(ping_array):
     vec = h.Vector(ping_array)
     stim.play(vec)
 
-    return stim, vec
\ No newline at end of file
+    return stim, vec
+
+
+def get_spine_number(section, density, area_density):
+    if not isinstance(section, Sec):
+        section = Sec(section)
+    if area_density:
+        area = section.hoc.L*np.pi*section.hoc.diam
+        spine_number = int(np.round(area * density))
+    else:
+        spine_number = int(np.round(section.hoc.L * density))
+
+    #if spine density is low (less than 1 per comp)
+    # use random number to determine whether to add a spine
+    if not spine_number:
+        rand = np.random.uniform()
+        if rand > density*section.hoc.L:
+            return 1
+        return 0
+    return spine_number
+
+
+def establish_electric_properties(section, spine_E_pas, spine_g_pas,
```

Comment:
* Add docstring
* As earlier - I think we should assume we are working with Sec wrapper rather than nrn.Section. In the whole neuron++ lib - I assume we are working with Sec object

Reply:
Docstring added.